src/provider/application.js

```diff
@@ -215,7 +215,10 @@ class Application extends EventEmitter {
       this.activeACL = origin;
     }
 
-    if (this.acls.includes('*') || this.acls.includes(origin)) {
+    if (this.acls.includes('*') || this.acls.includes(origin) || this.acls.some((acl) => {
+      const aclRegExp = new RegExp(acl.replace('*', '[a-zA-Z0-9-._]*'));
```

Comment:
This seems to fail if i do the following scenario
```js
const acl = "*.foo.com"
const aclRegExp = new RegExp(acl.replace('*', '[a-zA-Z0-9-._]*'));
"bar.foo.com.evil.com".match(aclRegExp);
```

I'm wondering if a safer approach is the following.
```
const subdomain = acl.replace(/^\*\/, ''); // Strip leading wildcard to get subdomain
return origin.endsWith(subdomain); // Verify origin is subdomain. Method needs polyfil for IE
```

Reply:
I like this idea.  I have made the changes and updated the documentation to specify the wildcard character is only supported as the first character of the string.